### PR TITLE
Hike index page

### PR DESCRIPTION
**Pull request by campercat**
Implemented the index page for all trails.

### CODE REVIEW

File: src/main/frontend/client/components/NavBar.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+const NavBar = () => {
+  return (
+    <div className="top-bar">
+      <div className="top-bar-left">
+        <ul className="menu">
+          <Link to="/trails">Trails Hub</Link>
+        </ul>
+      </div>
+    </div>
```
Comment (bkincart) [inline src/main/frontend/client/components/NavBar.js:12]: nicely done!

File: src/main/frontend/client/components/TrailsIndex.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  },[])
+
+  return(
+    <div>
+      <h1>Trail Index</h1>
+      {trailTiles}
+    </div>
+  )
+}
+
```
Comment (bkincart) [inline src/main/frontend/client/components/TrailsIndex.js:45]: Very nicely done! We could probably do with removing lines 6, 9, and 37 comments as they're not adding too much, but not a huge issue either way!

File: src/main/frontend/client/index.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    <link rel="stylesheet" href="./public/style.css">
+  </head>
+
+  <body>
+    <div id="app"></div>
+
+    <script src="//code.jquery.com/jquery-2.1.3.min.js"></script>
+    <script src="/bundle.js"></script>
+  </body>
+</html>
```
Comment (bkincart) [inline src/main/frontend/client/index.html]: We should not need this file, as the `src/main/resources/static/index.html` is doing this for us. Please remove!

File: src/main/java/com/launchacademy/reviews/models/Trail.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  @Column
+  private String description;
+
+  @NotNull
+  @DecimalMin("0.1")
+  @Column(columnDefinition = "numeric")
+  private Double distance;
+
+  @NotNull
+  @Min(0)
```
Comment (bkincart) [inline src/main/java/com/launchacademy/reviews/models/Trail.java:45]: Great! Nice validations throughout

File: src/main/java/com/launchacademy/reviews/services/TrailService.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  public Long count() {
+    return trailRepository.count();
+  }
+
+  public void save(Trail trail) {
+    trailRepository.save(trail);
+  }
+}
```
Comment (bkincart) [inline src/main/java/com/launchacademy/reviews/services/TrailService.java:29]: Nicely done here :)

**Review by bkincart — CHANGES_REQUESTED** [review]
A couple of changes to be made here:
- Please remove `src/main/frontend/client/public/images/test.png` and `src/main/frontend/client/index.html`
- Why was the sample migration file moved to be a `favicon.ico` file?

**Comment by campercat** [comment]
fixed requested changes

**Pull request merged**